src/net/hosts/store.rs

```diff
@@ -0,0 +1,1485 @@
+    //        // we will select from the whitelist.
+
+    //        let mut anchorlist = hosts.anchorlist.write().await;
+    //        anchorlist.clear();
+    //        drop(anchorlist);
+
+    //        let mut whitelist = hosts.whitelist.write().await;
+    //        whitelist.clear();
+    //        drop(whitelist);
+
+    //        let mut greylist = hosts.greylist.write().await;
+    //        greylist.clear();
+    //        drop(greylist);
+
+    //        assert!(hosts.is_empty_anchorlist().await);
+    //        assert!(hosts.is_empty_whitelist().await);
+    //        assert!(hosts.is_empty_greylist().await);
+
+    //        for i in 0..n_white {
+    //            white_urls.push(Url::parse(&format!("tcp://whitelist{}:123", i)).unwrap());
+    //        }
+    //        for i in 0..n_grey {
+    //            grey_urls.push(Url::parse(&format!("tcp://greylist{}:123", i)).unwrap());
+    //        }
+
+    //        for white in &white_urls {
+    //            let last_seen = UNIX_EPOCH.elapsed().unwrap().as_secs();
+    //            hosts.whitelist_store(white.clone(), last_seen).await;
+    //        }
+    //        for grey in &grey_urls {
+    //            let last_seen = UNIX_EPOCH.elapsed().unwrap().as_secs();
+    //            hosts.greylist_store(grey.clone(), last_seen).await;
+    //        }
+
+    //        assert!(hosts.is_empty_anchorlist().await);
+    //        assert!(!hosts.is_empty_whitelist().await);
+    //        assert!(!hosts.is_empty_greylist().await);
+
+    //        let host =
+    //            hosts.anchorlist_fetch_address(p2p.clone(), &transports).await.unwrap();
+
+    //        assert!(white_urls.iter().any(|u| u.clone() == host.0));
+
+    //        // If we have no whitelist connections, select from the greylist.
+    //        let mut whitelist = hosts.whitelist.write().await;
+    //        whitelist.clear();
+    //        drop(whitelist);
+
+    //        let mut greylist = hosts.greylist.write().await;
+    //        greylist.clear();
+    //        drop(greylist);
+
+    //        assert!(hosts.is_empty_anchorlist().await);
+    //        assert!(hosts.is_empty_whitelist().await);
+    //        assert!(hosts.is_empty_greylist().await);
+
+    //        for i in 0..n_grey {
+    //            grey_urls.push(Url::parse(&format!("tcp://greylist{}:123", i)).unwrap());
+    //        }
+
+    //        for grey in &grey_urls {
+    //            let last_seen = UNIX_EPOCH.elapsed().unwrap().as_secs();
+    //            hosts.greylist_store(grey.clone(), last_seen).await;
+    //        }
+
+    //        assert!(hosts.is_empty_anchorlist().await);
+    //        assert!(hosts.is_empty_whitelist().await);
+    //        assert!(!hosts.is_empty_greylist().await);
+
+    //        let host =
+    //            hosts.anchorlist_fetch_address(p2p.clone(), &transports).await.unwrap();
+
+    //        assert!(grey_urls.iter().any(|u| u.clone() == host.0));
+    //    });
+    //}
+}
```

Comment:
Use `/* */` style comments for big code blocks

Reply:
Or better yet, delete it all.

That's what version control is for, if you want to readd it later :-)

The commented out stuff is just more cognitive load for everyone.

Reply:
I simply forgot to reimplement the tests. Fixed on 639f1f72bf2908837ac1de0cd6baeef907a5ac96